docs/component-stability.md

```diff
@@ -66,6 +66,99 @@ Stable components MUST be compatible between minor versions unless critical secu
+Please check the documentation to verify which parts, if any, need to be implemented manually.
+
+**Definition:** In the following, an "item" refers generically to a single log record, metric point,
+or span.
+
+The internal telemetry of a stable pipeline component should allow observing the following:
+
+1. How much data the component receives.
+
+    For receivers, this could be a metric counting requests, received bytes, scraping attempts, etc.
+
+    For other components, this would typically be the number of items received through the
+    `Consumer` API.
+
+2. How much data the component outputs.
+
+    For exporters, this could be a metric counting requests, sent bytes, etc.
+
+    For other components, this would typically be the number of items forwarded to the next
+    component through the `Consumer` API.
+
+3. How much data is dropped because of errors.
+
+    For receivers, this could include a metric counting payloads that could not be parsed in.
+
+    For receivers and exporters that interact with an external service, this could include a metric
+    counting requests that failed because of network errors.
+
+    The goal is to be able to easily pinpoint the source of data loss in the Collector pipeline, so
+    this should either:
+    - only include errors internal to the component, or;
+    - allow distinguishing said errors from ones originating in an external service, or propagated
+        from downstream Collector components.
+
+4. Details for error conditions.
+
+    This could be in the form of logs or spans detailing the reason for an error. As much detail as
+    necessary should be provided to ease debugging. Processed signal data should not be included for
+    security and privacy reasons.
+
+5. Other possible discrepancies between input and output, if any. This may include:
+
+    - How much data is dropped as part of normal operation (eg. filtered out).
+
+    - How much data is created by the component.
+
+    - How much data is currently held by the component (eg. an UpDownCounter keeping track of the
+        size of an internal queue).
+
+6. Processing performance.
+
+    This could include spans for each operation of the component, or a histogram of end-to-end
+    component latency.
+    
+    The goal is to be able to easily pinpoint the source of latency in the Collector pipeline, so
+    this should either:
+    - only include time spent processing inside the component, or;
+    - allow distinguishing this latency from that caused by an external service, or from time spent
+        in downstream Collector components.
+
+When measuring amounts of data, counting items is recommended. Where this can't easily be done, any
+relevant unit may be used, as long as zero is a reliable indicator of the absence of data. In any
+case, all metrics should have a defined unit (not "1").
+
+If data can be dropped/created/held at multiple distinct points in a component's pipeline (eg.
```

Comment:
is there a recommendation about when it's okay to break a trace? Right now traces can be broken in components that do batching/queueing, and I think it would be useful to recommend that breaking the context should be avoided or a spanlink should be added.

Reply:
That sounds good. Following the recommendation that spans should "only include time spent processing inside the component", I think we should *recommend* having multiple spans when data is queued internally, but that the two spans must be either in the same trace (no breaking the context, which may be complex to implement), or have a link between them. I'll add that if it sounds good to you.

Reply:
yep, sounds reasonable. thank you!